devito/finite_differences/differentiable.py

```diff
@@ -304,8 +325,46 @@ class Mul(DifferentiableOp, sympy.Mul):
     __sympy_class__ = sympy.Mul
     __new__ = DifferentiableOp.__new__
 
+    @property
+    def _gather_for_diff(self):
+        """
+        We handle Mul arguments by hand in case of staggered inputs
+        such as `f(x)*g(x + h_x/2)`
+        In that case, priority of indexing is applied to have single indices
```

Comment:
the docstring was clear until this point. Then, as of this sentence, I'm a bit confused. I think a clearer example would help